src/main/java/assignment3/services/GraphService.java

```diff
@@ -0,0 +1,295 @@
+package assignment3.services;
+
+import assignment3.constants.Constants;
+import assignment3.exceptions.CyclicDependencyException;
+import assignment3.exceptions.DependencyNotFoundException;
+import assignment3.exceptions.NodeAlreadyExistsException;
+import assignment3.exceptions.NodeNotFoundException;
+import assignment3.models.GraphModel;
+import assignment3.models.NodeModel;
+import java.util.HashSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GraphService {
+
+  final static Logger logger = LoggerFactory.getLogger(GraphService.class);
+
+  /**
+   * Function which determines if node already exists in the graph
+   *
+   * @param id    ID of the node to insert
+   * @param nodes The graph
+   * @return True is node is already present, false otherwise
+   */
+  public boolean isNodePresentAlready(int id, GraphModel nodes) {
```

Comment:
Create one GraphUtility Class, which will do all these kind of operations. (irrespective of model class)